src/DeploymentProvider/DeploymentHelper.ts

```diff
@@ -100,29 +100,106 @@ export class DeploymentHelper {
+        core.debug('deploy response: ' + JSON.stringify(response));
+        return;
+    }
+
+    public static async deployEnterprise(client: asa.AppPlatformManagementClient, params: ActionParameters, sourceType: string, fileToUpload: string, resourceId: string) {
+        const buildServiceName = "default";
+        const buildName = `${params.appName}-${params.deploymentName}`;
+        core.debug('Starting deploy for enterprise');
+        const uploadResponse = await client.buildServiceOperations.getResourceUploadUrl(params.resourceGroupName, params.serviceName, buildServiceName);
+        core.debug('request upload url response: ' +  JSON.stringify(uploadResponse));
+        await uploadFileToSasUrl(uploadResponse.uploadUrl, fileToUpload);
+        const build: asa.Build = {
+            properties: {
+                relativePath: uploadResponse.relativePath,
+                builder: params.builder ? `${resourceId}/buildServices/${buildServiceName}/builders/${params.builder}` : `${resourceId}/buildServices/${buildServiceName}/builders/default`,
+                agentPool: `${resourceId}/buildServices/${buildServiceName}/agentPools/default`,
+            }
+        };
+        let transformedBuildEnvironmentVariables = {};
+        if (params.buildEnv) {
+            core.debug("Build environment variables modified.");
+            const parsedBuildEnvVariables = parse(params.buildEnv);
+            //Parsed pairs come back as  {"key1":{"value":"val1"},"key2":{"value":"val2"}}
+            Object.keys(parsedBuildEnvVariables).forEach(key => {
+                transformedBuildEnvironmentVariables[key] = parsedBuildEnvVariables[key]['value'];
+            });
+            build.properties.env = transformedBuildEnvironmentVariables;
+        }
+        core.debug('build: ' +  JSON.stringify(build));
+        const buildResponse = await client.buildServiceOperations.createOrUpdateBuild(params.resourceGroupName, params.serviceName, buildServiceName, buildName, build);
+        core.debug('build response: ' +  JSON.stringify(buildResponse));
+        const regex = RegExp("[^/]+$");
+        const buildResultName = regex.exec(buildResponse.properties.triggeredBuildResult.id)[0];
+        let buildProvisioningState = 'Queuing';
+        let cnt = 0;
+        core.debug("wait for build result......");
+        //Waiting for build result. Timeout 30 minutes.
+        while (buildProvisioningState != 'Succeeded' && buildProvisioningState != 'Failed' && cnt++ < 180) {
+            await new Promise(resolve => setTimeout(resolve, 10000));
+            console.log("wait for 10 seconds....");
+            const waitResponse = await client.buildServiceOperations.getBuildResult(params.resourceGroupName, params.serviceName, buildServiceName, buildName, buildResultName);
+            core.debug('build result response: ' + JSON.stringify(waitResponse));
```

Comment:
It's likely we are going to generate a lot of messages here. Do we have any different, useful information in each call?

Reply:
The messages are only printed when debug mode on. I think the debug mode only on for internal or customers encounter any problems. The debug messages may help us to address the problem. For this case, it will show the current buildStages.
![image](https://user-images.githubusercontent.com/86712051/190550938-fb8e7c66-9708-4f0f-a6c2-ce20bfaeb7e6.png)